lib/cc/config/prepare.rb

```diff
@@ -0,0 +1,68 @@
+module CC
+  module Config
+    class Prepare
+      class NoPrepareNeeded
+        def fetch
+          []
+        end
+      end
+
+      attr_reader :fetch
+
+      def self.from_yaml(data)
+        if data.present?
+          fetch = Fetch.from_yaml(data.fetch("fetch", []))
+
+          new(fetch: fetch)
+        else
+          NoPrepareNeeded.new
+        end
+      end
+
+      def initialize(fetch:)
+        @fetch = fetch
+      end
+
+      class Fetch
+        def self.from_yaml(data)
```

Comment:
Yeah, that's where I figured we'd end up. I was more thinking that we don't want to totally divorce that validate logic from the data munging in these ``CC::Config::*` classes. So maybe they'll raise exceptions or set error flags & the top-level config building runner will catch/use those and `validate-config` just builds the config & echos errors?